### DISCUSSION TITLE
Can someone direct me to the function that takes the JSONL file from tavern AI and feeds it to the model

### DISCUSSION THREAD

**Discussion by PalaSerialX** [oobabooga/text-generation-webui#1506] [Q&A] [answered]
as the title says, I am trying to find the function located that polls the GET/POST request from tavern AI to get the JSONL file and then write the reply into that same file. Please.

**Comment by bmoconno** [comment, marked as answer]
I think you're probably looking for the [api ](https://github.com/oobabooga/text-generation-webui/tree/main/extensions/api) extension, then based on if you're using blocking or streaming api you'll either check out the [blocking_api.py](https://github.com/oobabooga/text-generation-webui/blob/main/extensions/api/blocking_api.py) or [streaming_api.py](https://github.com/oobabooga/text-generation-webui/blob/main/extensions/api/streaming_api.py)

**Reply by PalaSerialX** [reply]
Thanks, I did find it. It was in API extensions, when I loaded it into pycharm. it was inside script.py. I was able to see the flow from the API to the generate_reply function and can now make the mods i need. I appreciate the help!
